enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java

```diff
@@ -0,0 +1,201 @@
+import static com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus.*;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Document verification service providing {@link #executeFinalDocumentVerification(IdentityVerificationEntity, OwnerId)}.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+@Service
+@Slf4j
+public class DocumentVerificationService {
+
+    private final DocumentVerificationProvider documentVerificationProvider;
+
+    private final IdentityVerificationService identityVerificationService;
+
+    private final CommonOnboardingService processService;
+
+    private final OnboardingProcessLimitService processLimitService;
+
+    private final AuditService auditService;
+
+    @Autowired
+    public DocumentVerificationService(
+            final DocumentVerificationProvider documentVerificationProvider,
+            final IdentityVerificationService identityVerificationService,
+            final CommonOnboardingService processService,
+            final OnboardingProcessLimitService processLimitService,
+            final AuditService auditService) {
+
+        this.documentVerificationProvider = documentVerificationProvider;
+        this.identityVerificationService = identityVerificationService;
+        this.processService = processService;
+        this.processLimitService = processLimitService;
+        this.auditService = auditService;
+    }
+
+    /**
+     * Execute final document verification of the given identity verification.
+     * <p>
+     * Based on the result of calling document verification provider, change the identity verification status.
+     * Also change status of the document verifications accordingly.
+     *
+     * @param identityVerification Identification verification whose documents should be verified
+     * @param ownerId Owner identification
+     * @throws RemoteCommunicationException In case of remote communication error
+     * @throws DocumentVerificationException In case of business logic error
+     * @throws OnboardingProcessException When process not found
+     */
+    public void executeFinalDocumentVerification(final IdentityVerificationEntity identityVerification, final OwnerId ownerId)
+            throws RemoteCommunicationException, DocumentVerificationException, OnboardingProcessException {
+
+        identityVerificationService.moveToPhaseAndStatus(identityVerification, DOCUMENT_VERIFICATION_FINAL, IN_PROGRESS, ownerId);
+
+        final List<DocumentVerificationEntity> documentVerifications = filterDocumentVerifications(identityVerification);
+
+        final List<String> uploadIds = documentVerifications.stream()
+                .map(DocumentVerificationEntity::getUploadId)
+                .collect(toList());
+
+        final DocumentsVerificationResult result = documentVerificationProvider.verifyDocuments(ownerId, uploadIds);
```

Comment:
In case of an error during `verifyDocuments` call the identity verification may get stuck in this phase/status. Let's raise a ticket for handling this separately.

Reply:
Raised as: https://github.com/wultra/enrollment-server/issues/496